src/viam/app/data_client.py

```diff
@@ -220,6 +239,95 @@ def resource_subtype(self) -> str:
             )
             return self.resource_api
 
+
+    @dataclass
+    class DataPipeline:
+        """Represents a data pipeline and its associated metadata."""
+
+        id: str
+        """The ID of the data pipeline"""
+
+        organization_id: str
+        """The organization ID"""
+
+        name: str
+        """The name of the data pipeline"""
+
+        mql_binary: List[Dict[str, Any]]
+        """The MQL binary of the data pipeline"""
+
+        schedule: str
+        """The schedule of the data pipeline"""
+
+        created_on: datetime
+        """The time the data pipeline was created"""
+
+        updated_at: datetime
+        """The time the data pipeline was last updated"""
+
+        enabled: bool
+        """Whether the data pipeline is enabled"""
+
+    @dataclass
+    class DataPipelineRun:
+        """Represents a data pipeline run and its associated metadata."""
+        id: str
+        """The ID of the data pipeline run"""
+
+        status: DataPipelineRunStatus.ValueType
+        """The status of the data pipeline run"""
+
+        start_time: datetime
+        """The time the data pipeline run started"""
+
+        end_time: datetime
+        """The time the data pipeline run ended"""
+
+        data_start_time: datetime
+        """The start time of the data that was processed in the run."""
+        data_end_time: datetime
+        """The end time of the data that was processed in the run."""
+
+    @dataclass
+    class DataPipelineRunsPage:
+        """Represents a page of data pipeline runs and provides pagination functionality."""
+
+        _client: "DataClient"
+        """The data client used to make API calls"""
+
+        pipeline_id: str
+        """The ID of the pipeline these runs belong to"""
+
+        page_size: int
+        """The number of runs per page"""
+
+        runs: List["DataClient.DataPipelineRun"]
+        """The list of runs in this page"""
+
+        next_page_token: str
+        """The token to use to get the next page of results"""
+
+        async def next_page(self) -> "DataClient.DataPipelineRunsPage":
+            """Get the next page of data pipeline runs.
+
+            Returns:
+                DataPipelineRunsPage: The next page of runs, or an empty page if there are no more runs
+            """
+            if not self.next_page_token:
+                # no token, return empty next page
+                return DataClient.DataPipelineRunsPage(
+                    _client=self._client,
+                    pipeline_id=self.pipeline_id,
+                    page_size=self.page_size,
+                    runs=[],
+                    next_page_token=""
+                )
+            return await self._client._list_data_pipeline_runs(
+                self.pipeline_id,
+                self.page_size,
+                self.next_page_token
+            )
```

Comment:
All of these shadow classes should have a `from_proto` class method for reading, and a `proto` property if they are for writing. See [here](https://github.com/viamrobotics/viam-python-sdk/blob/fb19be2169d2c5b17f38cc32fdf98eeb0fe5a068/src/viam/app/app_client.py#L172) for an example